sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -506,6 +506,17 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(true)
 
+  val SQL_EXPLAIN_ONLY_ENABLED = conf("spark.rapids.sql.explainOnly.enabled")
```

Comment:
Since this essentially disables running on the GPU, would it make more sense to extend the existing `spark.rapids.sql.enabled` property to allow an 'explain' or `explainOnly` value that means it's only enabled for explanation only?  Otherwise you get into a situation where `explainOnly` is enabled but `spark.rapids.sql.enabled` is `false`, and then what would a user expect to happen?

Reply:
we certainly can, was thinking changing from boolean might be weird since to me "enabled"is generally a true or false setting.  Other possibilities would be change this to be spark.rapids.sql.mode where values are explain or convert (or something similar), then the enabled = false would truly disable it no matter the mode.
thoughts?

Reply:
the other option I had considered was splitting it out more so you didn't specify the spark.plugins=com.nvidia.spark.SQLPlugin... if we are ever able to separate the code enough to not need cudf and such that might make more sense.

Reply:
discussed offline and going to hold off a bit on deciding between the mode config and the separate plugin to investigate Spark 2.0 explain api.

Reply:
the current 2.x implementation uses a completely separate jar and no configs are required, the user just calls an API. So from that perspective I'm not sure how this is implemented compared to that matters since no configuration needed.  

I'm not sure I have a strong opinion on either of the proposals. A few of the pros and cons:
Using just the config like spark.rapids.sql.mode will be less code changes, in some ways it might be easier to flip back and forth between the modes as well, but it won't be able to be changed run time because of the initialization differences. So in that aspect it could be confusing.
If we go with separate plugin it might be less likely to accidentally break if someone changes things in current code and doesn't think about this code.  Here I'm talking about things like accidentally initializing gpu or shuffle in this mode.
if we are able to separate out and have just a jar for tagging logic, that can be good and bad, less to download if just doing the explain but yet another jar user might need to deal with and if they switch to gpu nodes need to setup it up differently.

If anyone has other thoughts or opinions let me know. Otherwise I might go with the 'spark.rapids.sql.mode' config as its less change.